src/chronify/time_series_mapper_representative.py

```diff
@@ -0,0 +1,129 @@
+            dfm = self._generator._handler.add_time_attribute_columns(dft, to_time_col)
+        else:
+            with self._engine.connect() as conn:
+                table = Table(self.from_schema.name, self._metadata)
+                stmt = (
+                    select(table.c["time_zone"])
+                    .distinct()
+                    .where(table.c["time_zone"].is_not(None))
+                )
+                time_zones = read_database(stmt, conn, self.from_schema.time_config)[
+                    "time_zone"
+                ].to_list()
+            dfm = self._generator._handler.create_tz_aware_mapping_dataframe(
+                dft, to_time_col, time_zones
+            )
+
+        # Ingest mapping into db
+        time_array_id_columns = [
+            x for x in self.from_schema.time_config.list_time_columns() if x != "hour"
+        ]
+        if not is_tz_naive:
+            time_array_id_columns += ["time_zone"]
+
+        map_table_name = "map_table"
+        with self._engine.connect() as conn:
+            write_database(
+                dfm, conn, map_table_name, self.to_schema.time_config, if_table_exists="replace"
+            )
+            conn.commit()
+        self._metadata.reflect(self._engine, views=True)
+
+        self._apply_mapping(map_table_name)
+
+    def _apply_mapping(self, map_table_name: str):
+        """Apply mapping to create result as a view according to_schema"""
+        left_table = Table(self.from_schema.name, self._metadata)
+        right_table = Table(map_table_name, self._metadata)
+        left_table_columns = [x.name for x in left_table.columns]
+        right_table_columns = [x.name for x in right_table.columns]
+
+        final_cols = self.to_schema.list_columns()
+        left_cols = [x for x in left_table_columns if x in final_cols]
+        right_cols = [x for x in right_table_columns if x in final_cols and x not in left_cols]
+        assert set(left_cols + right_cols) == set(
+            final_cols
+        ), f"table join does not produce the {final_cols=}"
+
+        select_stmt = [left_table.c[x] for x in left_cols]
+        select_stmt += [right_table.c[x] for x in right_cols]
+
+        keys = self.from_schema.time_config.list_time_columns()
+        if not self.to_schema.time_config.is_time_zone_naive():
+            keys += ["time_zone"]
+            assert (
+                "time_zone" in left_table_columns
+            ), f"time_zone not in table={self.from_schema.name}"
+            assert "time_zone" in right_table_columns, f"time_zone not in table={map_table_name}"
+        on_stmt = ()
+        for i, x in enumerate(keys):
+            if i == 0:
+                on_stmt = left_table.c[x] == right_table.c[x]
+            else:
+                on_stmt &= left_table.c[x] == right_table.c[x]
+        query = select(*select_stmt).select_from(left_table).join(right_table, on_stmt)
+        create_view(self.to_schema.name, query, self._engine, self._metadata)
```

Comment:
Make into table creation cmd, need conn obj. If fails call rollback.

Reply:
[1] modify sqlachemy_view funcs to accept table as a flag.
[2] if [1] fails, call sql cmd `create table as select * from view`.